torch/_dynamo/variables/builder.py

```diff
@@ -2742,7 +2742,9 @@ def update_dim2constraint(dim, constraint_range, name):
         constraint_sizes.append(constraint_size)
         constraint_strides.append(constraint_stride)
 
-        if marked_unbacked:
+        if marked_unbacked and e._dynamo_assume_contiguous:
+            dynamic_size = DimDynamic.SIZE_LIKE_UNBACKED_CONTIGUOUS
```

Comment:
This (having a new SIZE_LIKE_UNBACKED_CONTIGUOUS) is sus, but I need to read more.